File: backend/agents.py


ROOMS: dict[str, dict] = {
    "pm_zone":    {"x": 68,  "y": 60},
    "dev_zone":   {"x": 580, "y": 60},
    "meeting":    {"x": 330, "y": 270},
    "qa_zone":    {"x": 68,  "y": 420},
    "design":     {"x": 580, "y": 390},
    "ceo_office": {"x": 580, "y": 480},
}

AGENT_CONFIGS = [
    {"id": "ceo",      "role": "CEO",      "emoji": "👔", "color": "#eab308", "home": "ceo_office"},
    {"id": "pm",       "role": "PM",       "emoji": "🧠", "color": "#6366f1", "home": "pm_zone"},
    {"id": "dev",      "role": "Dev",      "emoji": "⚙️", "color": "#0ea5e9", "home": "dev_zone"},
    {"id": "qa",       "role": "QA",       "emoji": "🔬", "color": "#a855f7", "home": "qa_zone"},
    {"id": "designer", "role": "Designer", "emoji": "🎨", "color": "#22c55e", "home": "design"},
    {"id": "devops",   "role": "DevOps",   "emoji": "🚀", "color": "#f97316", "home": "meeting"},
]


def build_initial_agents() -> dict[str, dict]:
    agents = {}
    for cfg in AGENT_CONFIGS:
        room_key = cfg["home"]
        coords = ROOMS[room_key]
        agents[cfg["id"]] = {
            "id":      cfg["id"],
            "role":    cfg["role"],
            "emoji":   cfg["emoji"],
            "color":   cfg["color"],
            "room":    room_key,
            "x":       coords["x"],
            "y":       coords["y"],
            "status":  "idle",
            "message": "",
        }
    return agents

File: backend/agents.ts
import type { AgentId } from "./types"

export interface AgentDef {
  id: AgentId
  role: string
  color: string
  homeRoom: string
  gstackSkill: string
}

export interface RoomDef {
  id: string
  label: string
  position: [number, number, number]
}

export const ROOMS: RoomDef[] = [
  { id: "ceo_office",  label: "CEO 오피스", position: [0,   0, -8] },
  { id: "pm_zone",     label: "PM 존",      position: [-6,  0, -4] },
  { id: "meeting",     label: "미팅룸",     position: [0,   0,  0] },
  { id: "dev_zone",    label: "Dev 존",     position: [6,   0, -4] },
  { id: "qa_zone",     label: "QA 존",      position: [-6,  0,  4] },
  { id: "design_zone", label: "Design 존",  position: [6,   0,  4] },
  { id: "devops_zone", label: "DevOps 존",  position: [0,   0,  8] },
]

export const AGENTS: AgentDef[] = [
  { id: "ceo",      role: "CEO",      color: "#eab308", homeRoom: "ceo_office",  gstackSkill: "agent-ceo" },
  { id: "pm",       role: "PM",       color: "#6366f1", homeRoom: "pm_zone",     gstackSkill: "agent-pm" },
  { id: "designer", role: "Designer", color: "#22c55e", homeRoom: "design_zone", gstackSkill: "agent-designer" },
  { id: "dev",      role: "Dev",      color: "#0ea5e9", homeRoom: "dev_zone",    gstackSkill: "agent-dev" },
  { id: "qa",       role: "QA",       color: "#a855f7", homeRoom: "qa_zone",     gstackSkill: "agent-qa" },
  { id: "devops",   role: "DevOps",   color: "#f97316", homeRoom: "devops_zone", gstackSkill: "agent-devops" },
]

export function getAgentDef(id: string): AgentDef | undefined {
  return AGENTS.find(a => a.id === id)
}

export function getRoomPosition(roomId: string): [number, number, number] {
  return ROOMS.find(r => r.id === roomId)?.position ?? [0, 0, 0]
}
